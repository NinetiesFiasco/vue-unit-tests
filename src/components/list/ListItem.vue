<template>
  <div>
    <h2>{{title}}</h2>
    <p>{{description}}</p>
  </div>
</template>

<script>
export default {
  name: 'ListItem',
  props: {
    title: {type: String, default: ''},
    description: {type: String, default: ''}
  }
}
</script>

<style>

</style>