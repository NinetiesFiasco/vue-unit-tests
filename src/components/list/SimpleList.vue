<template>
  <div>
    <template v-for="(item, key) of list" :key="key">
      <ListItem :title="item.title" :description="item.description" />
    </template>

    <div>
      Here we have two tests:
      <ul>
        <li>First test that we have rendered component ListItem with one entry with given title and description </li>
        <li>Second test that we have rendered component SimpleList with one entry of ListItem and we test just count</li>
      </ul>
      FYI
      <ul>
        <li>shallowMount renders only component without children, other way mount renders component with children </li>
      </ul>
    </div>
  </div>
</template>

<script>
import ListItem from './ListItem.vue'

export default {
  name: 'SimpleList',
  components: {ListItem},
  props: {
    list: {type: Array, default: () => [{
      title: 'my title',
      description: 'my description'
    },{
      title: 'my title 2',
      description: 'my description 2'
    }]}
  }
}
</script>

<style>

</style>