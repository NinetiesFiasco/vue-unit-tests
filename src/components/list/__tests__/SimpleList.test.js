import { shallowMount, mount } from '@vue/test-utils'
import SimpleList from '../SimpleList.vue'
import ListItem from '../ListItem.vue'

test('SimpleList renders the right number of ListItem', () => {
  const wrapper = shallowMount(SimpleList, {
    propsData: {
      list: [{
        title: "my title",
        description: "my description"
      }]
    }
  })

  expect(wrapper.findAllComponents(ListItem)).toHaveLength(1) 
})

test('SimpleList renders passed title and description for each passed ListItem', () => {
  const wrapper = mount(SimpleList, {
    propsData: {
      list: [{
        title: "my title",
        description: "my description"
      },{
        title: "my title 2",
        description: "my description 2"
      }]
    }
  })

  const outputText = wrapper.text()
  expect(outputText).toContain('my title')
  expect(outputText).toContain('my description')
  expect(outputText).toContain('my title 2')
  expect(outputText).toContain('my description 2')
})

test('Snapshot test', () => {
  const wrapper = mount(SimpleList, {
    propsData: {
      list: [{
        title: "my title 3",
        description: "my description 3"
      },{
        title: "my title 4",
        description: "my description 4"
      }]
    }
  })

  expect(wrapper.text()).toMatchSnapshot()
})