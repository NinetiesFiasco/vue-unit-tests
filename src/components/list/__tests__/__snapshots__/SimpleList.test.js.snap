// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `"my title 3my description 3my title 4my description 4 Here we have two tests: First test that we have rendered component ListItem with one entry with given title and description Second test that we have rendered component SimpleList with one entry of ListItem and we test just count FYI shallowMount renders only component without children, other way mount renders component with children"`;
